src/Mutator/Arithmetic/Assignment.php

```diff
@@ -9,26 +9,24 @@
 
 namespace Infection\Mutator\Arithmetic;
 
-use Infection\Mutator;
-use Infection\Mutator\Util\Replacer\MutatorResult;
-use Infection\Mutator\Util\Replacer\SingleMutationResult;
+use Infection\Mutator\Util\SingleMutator;
 use PhpParser\Node;
 
 /**
  * @internal
  */
-final class Assignment extends Mutator
+final class Assignment extends SingleMutator
 {
     /**
      * Replaces "+=", "*=", ".=", and similar with a plain "="
      *
      * @param Node|Node\Expr\AssignOp $node
      *
-     * @return MutatorResult
+     * @return Node
      */
-    public function mutate(Node $node): MutatorResult
+    public function getMutatedNode(Node $node): Node
```

Comment:
This should be `protected`, not `public`. 

`abstract protected function getMutatedNode(Node $node): array;` in the parent

Same for all classes.